cmd/main.go

```diff
@@ -382,7 +382,7 @@ func main() {
 		failOnError(cerr, "Failed to create kubernetes cluster config")
 		k8sClient = kubernetes.NewForConfigOrDie(cfg)
 
-		autoMigrationLeader = leader.NewElector(k8sClient, leader.Config{LeaseDuration: 5 * time.Second,
+		secondLeader = leader.NewElector(k8sClient, leader.Config{LeaseDuration: 5 * time.Second,
```

Comment:
I'm not sure reusing this is the best option. Now querying for release images will prevent migration from running.

Maybe a better solution on startup would be to use a database lock on the table.
Then check if the table is empty, if it is populate it and release the table lock, if not release the lock with no action.

This would lead to only one replica populating the table on first deploy then later updates would be done from the thread which uses the "regular" leader.

Reply:
I think this solution is a good fit for now for patching stage, then we can improve the solution by trying to use postgres resource protection mechanism instead

Reply:
Okay, if this is actually going to be temporary then it's fine for now, but please open an issue to revisit this.